datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+        "type": "person",
+    },
+    # {  # child kidnappers
+    #     "url": "https://war-sanctions.gur.gov.ua/en/kidnappers/companies?page=1&per-page=12",
+    #     "type": "company",
+    # },
+    {  # russian athletes
+        "url": "https://war-sanctions.gur.gov.ua/en/sport/persons?page=1&per-page=12",
+        "type": "person",
+    },
+    # {  # ships
+    #     "url": "https://war-sanctions.gur.gov.ua/en/transport/ships?page=1&per-page=12",
+    #     "type": "vessel",
+    # },
+]
+
+# TODO: fix strings that were merged on | but were not supposed to be
+
+
+def lookup_override(context, key, lookup_type):
+    override_res = context.lookup("overrides", key)
+    if not override_res:
+        context.log.warning(f"No override found for {key}")
+        return {}
+
+    extracted = {}
+    for override in override_res.items:
+        if override["prop"] == "name":
+            extracted["name"] = override["value"]
+        elif override["prop"] == "registrationCode":
+            extracted["registrationCode"] = override["value"]
+
+    if not extracted:
+        context.log.warning(f"No matching properties found in overrides for {key}")
+
+    return extracted
+
+
+def extract_label_value_pair(label_elem, value_elem, data):
+    label = label_elem.text_content().strip().replace("\n", " ")
+    value = value_elem.text_content().strip().replace("\n", " ")
+    value = " ".join(value.split())
+    # we use it for splitting some strings at a later stage
+    value = " | ".join(
+        [text.strip() for text in value_elem.itertext() if text.strip()]
+    ).strip()
+    data[label] = value
+
+    return data
+
+
+def crawl_index_page(context: Context, index_page, data_type):
+    index_page = context.fetch_html(index_page, cache_days=3)
+    main_grid = index_page.find('.//div[@id="main-grid"]')
+    if main_grid is not None:
+        for a in main_grid.findall(".//a"):
+            href = [a.get("href")]
+            for link in href:
+                if link.startswith("https:"):
+                    detail_page = context.fetch_html(link, cache_days=3)
+                    if data_type == "person":
+                        details_container = detail_page.find(
+                            ".//div[@id='js_visibility'][@class='col-12 col-lg-9']"
+                        )
+                        crawl_person(context, details_container, link)
+                    elif data_type == "company":
+                        details_container = detail_page.find(
+                            ".//div[@class='col-12 col-lg-9']"
+                        )
+                        crawl_company(context, details_container, link)
+                    if data_type == "vessel":
+                        details_container = detail_page.find(
+                            ".//div[@id='js_visibility']"
+                        )
+                        crawl_vessel(context, details_container, link)
```

Comment:
I wouldn't fetch and extract data from detail pages inside crawl_index_page. Let's move the fetch and container extraction to the respective detail functions, then 

- crawl_index_page will be focused on index pages
- crawl_ for detail pages will be focused on detail pages

let's also try and find more semantic selectors.

e.g. for vessel, js_visibility is about showing/hiding stuff using javascript. They might add more instances of it unrelated to the content we know we can handle. "col-12 col-lg-9" are about column width and responsiveness. I think just selecting //main will do the right thing if you then look for //section//tools-* below that later, right?